# Count mapped reads using STAR and XL genome

```
#!/bin/sh
#SBATCH --job-name=STAR_count
#SBATCH --nodes=4
#SBATCH --ntasks-per-node=1
#SBATCH --time=3:00:00
#SBATCH --mem=4Gb
#SBATCH --output=STAR_count.%J.out
#SBATCH --error=STAR_count.%J.err
#SBATCH --account=rrg-ben

# sbatch 2022_STAR_count.sh inputbam output_counts

module load StdEnv/2020 gcc/9.3.0 star/2.7.9a samtools subread/2.0.3
module load StdEnv/2023 star/2.7.11b

# must use -s 0 because the data are unstranded
# must use -p because the data are paired
# use --countReadPairs to count read pairs instead of reads
# use -C to prevent counting of chimeric reads
# -T is the number of threads
featureCounts -T 4 -s 0 -p --countReadPairs -C \
  -a /home/ben/projects/rrg-ben/ben/2021_XL_v10_refgenome/XENLA_10.1_GCF_XBmodels.gtf \
  -o ${2} \
  ${1}
```
